_targets.R

```diff
@@ -165,13 +157,73 @@ targets_list <- list(
   ##### Munge GDP output into NetCDF files that will feed into GLM #####
 
   # Munge GCM variables into useable GLM variables and correct units
+  # For each tile-gcm combo this function returns a list with 2 elements:
+  # 1) file_out - the name of the munged output file, which contains data
+  # only for queried cells in that tile that returned data for all variables, and
+  # 2) cell_info - a tibble with a row for each queried cell in that tile, indicating
+  # whether or not that cell is missing data for *any* variable for that gcm
+  tar_target(
+    glm_ready_gcm_data_list,
+    munge_notaro_to_glm(gcm_data_raw_feather, gcm_names, unique(query_cells_centroids_list_by_tile$tile_no)),
+    pattern = map(gcm_data_raw_feather, cross(query_cells_centroids_list_by_tile, gcm_names, gcm_dates_df))
+  ),
+
+  # Combine the `file_out` elements of the gcm_ready_gcm_data_list branches
+  # into a single vector of file targets
   tar_target(
     glm_ready_gcm_data_feather,
-    munge_notaro_to_glm(gcm_data_raw_feather),
-    pattern = map(gcm_data_raw_feather),
+    glm_ready_gcm_data_list$file_out,
+    pattern = map(glm_ready_gcm_data_list),
     format = "file"
   ),
 
+  # Combine the `cell_info` elements of the gcm_ready_gcm_data_list branches
+  # into a single tibble of cell information, with a row per cell-gcm combo
+  tar_target(
+    glm_ready_gcm_data_cell_info,
+    glm_ready_gcm_data_list$cell_info,
+    pattern = map(glm_ready_gcm_data_list),
+  ),
+
+  # Pivot the cell_info tibble wider so that we have a single row per cell
+  # and can track for how many gcms each cell is or is not missing data
+  tar_target(
+    glm_ready_gcm_data_cell_status,
+    glm_ready_gcm_data_cell_info %>%
+      pivot_wider(names_from='gcm', values_from='missing_data', names_glue="{gcm}_missing_data") %>%
+      mutate(n_gcm_missing_data = rowSums(across(c(-cell_no,-tile_no)))),
+  ),
+
+  # Re-do mapping of which lakes are in which cells and tiles, using only the query
+  # cells that returned data and did not have missing data for any variables for any GCMs.
+  # Here, each lake is matched to the cell with the centroid that is closest to the
+  # lake centroid. By using only those cells that returned data, we can ensure that lakes
+  # that fell within cells that were missing data are re-assigned to the closest
+  # cell that returned data. This xwalk is being used in `lake-temperature-process-models`
+  # to determine what meteorological data to pull for each lake.
+  tar_target(lake_cell_tile_xwalk_data_df,
+             get_lake_cell_tile_data_xwalk(query_lake_centroids_sf, query_cell_centroids_sf, glm_ready_gcm_data_cell_status)
+             ),
+
+  # Save the revised lake-cell-tile mapping for use in `lake-temperature-process-models`
+  tar_target(lake_cell_tile_xwalk_csv, {
+    out_file <- "7_drivers_munge/out/lake_cell_tile_xwalk.csv"
+    write_csv(lake_cell_tile_xwalk_data_df, out_file)
```

Comment:
Per your point about having two xwalks vs one with two columns, I think it matters most at this stage - what do we save out from the pipeline and share/use downstream? I think sharing just this xwalk doesn't make it clear to downstream users that we adjusted the lake to cell mapping. So, I think we should have two different columns for the `cell_no_spatial` and `cell_no_data` for each lake to make it clear that we redid some of the mapping.